vavr/src/main/java/io/vavr/collection/LinkedHashSet.java

```diff
@@ -485,11 +485,27 @@ private LinkedHashSet(LinkedHashMap<T, Object> map) {
         return LinkedHashSet.ofAll(Iterator.rangeClosedBy(from, toInclusive, step));
     }
 
+    /**
+     * Add the given element to this set, replacing existing one if it is already contained.
+     * <p>
+     * Note that this method has a worst-case linear complexity.
+     *
+     * @param element The element to be added.
+     * @return A new set containing all elements of this set and also {@code element}.
+     */
     @Override
     public LinkedHashSet<T> add(T element) {
         return contains(element) ? this : new LinkedHashSet<>(map.put(element, element));
     }
 
+    /**
+     * Adds all of the given elements to this set, replacing existing one if they are not already contained.
+     * <p>
+     * Note that this method has a worst-case quadratic complexity.
```

Comment:
Not applicable to this PR, but it seems that we can do this in linear time. `LinkedHashMap` consists of two objects: `List` and `HashMap`. In fact, we should two things here:
* maybe to remove `n` elements from `HashMap` and add `n` elements to `HashMap`. This can be done in linear time.
* maybe to remove `n` elements from `List` and add `n` elements to `List`. This also can be done in linear time. Something like this: `list.filter(additional::contains).appendAll(additional)`

Reply:
That would _slightly_ change the current behavior, that the elements in the  `List` are kept in the order they were _first inserted_ to the `List`, and thus to the `LinkedHashMap` and `LinkedHashSet` it is backing. Whether that is acceptable or even desirable, that I don't know ⛄️

Reply:
Yep, sorry :) Very old story.